incubator/hnc/internal/reconcilers/anchor.go

```diff
@@ -135,17 +143,36 @@ func (r *AnchorReconciler) onDeleting(ctx context.Context, log logr.Logger, inst
 	}
 }
 
+// isDeletingCRD returns true if the Anchor CRD is being deleted. This can happen if HNC is being
+// uninstalled. In such cases, we shouldn't perform a cascading deletion.
+func (r *AnchorReconciler) isDeletingCRD(ctx context.Context) (bool, error) {
+	crd := &apiextensions.CustomResourceDefinition{}
+	nsn := types.NamespacedName{Name: api.Anchors + "." + api.MetaGroup}
+	if err := r.Get(ctx, nsn, crd); err != nil {
+		// Either the CRD wasn't found, in which case, HNC can't operate propertly and is hopefully
+		// being deleted itself. Otherwise, there was some transient error and we should just retry.
```

Comment:
For the first sentence, does it mean the CRD is already removed? What does "is hopefully being deleted itself" mean?

Reply:
Updated comment